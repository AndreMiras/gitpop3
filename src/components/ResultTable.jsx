import React from 'react';
import PropTypes from 'prop-types';
import { Table } from 'react-bootstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Pagination from 'react-js-pagination';
import ForkLine from './ForkLine';

const paginatedForks = (forks, activePage, itemsCountPerPage) => (
  forks.slice((activePage - 1) * itemsCountPerPage, activePage * itemsCountPerPage)
);

const ResultTable = ({
  forks, activePage, itemsCountPerPage, onPageChange,
}) => (
  <>
    <Table striped bordered hover>
      <thead>
        <tr>
          <th>
            <FontAwesomeIcon icon={['fab', 'github-alt']} />
            {' '}
            Repo
          </th>
          <th>
            <FontAwesomeIcon icon="star" />
            {' '}
            Stars
          </th>
          <th>
            <FontAwesomeIcon icon="code-branch" />
            {' '}
            Forks
          </th>
          <th>
            <FontAwesomeIcon icon="dot-circle" />
            {' '}
            Commits
          </th>
          <th>
            <FontAwesomeIcon icon="calendar-alt" />
            {' '}
            Modified
          </th>
        </tr>
      </thead>
      <tbody>
        {
          paginatedForks(
            forks,
            activePage,
            itemsCountPerPage,
          ).map(
            (fork) => <ForkLine key={fork.nameWithOwner} info={fork} />,
          )
        }
      </tbody>
    </Table>
    <Pagination
      itemClass="page-item"
      linkClass="page-link"
      activePage={activePage}
      itemsCountPerPage={itemsCountPerPage}
      totalItemsCount={forks.length}
      pageRangeDisplayed={5}
      onChange={onPageChange}
    />
  </>
);
ResultTable.propTypes = {
  forks: PropTypes.arrayOf(ForkLine.propTypes.info).isRequired,
  activePage: PropTypes.number.isRequired,
  itemsCountPerPage: PropTypes.number.isRequired,
  onPageChange: PropTypes.func.isRequired,
};

export default ResultTable;
