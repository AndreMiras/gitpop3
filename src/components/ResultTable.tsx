import { FunctionComponent, useState } from "react";
import { orderBy as lodashOrderBy } from "lodash";
import { OverlayTrigger, Table, Tooltip } from "react-bootstrap";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import Pagination from "react-js-pagination";
import { Node } from "../utils/types";
import ForkLine from "./ForkLine";

type Direction = "desc" | "asc";

type SortFunc = (collection: any, order: Direction) => any;

const paginatedForks = (
  forks: Node[],
  activePage: number,
  itemsCountPerPage: number
) =>
  forks.slice(
    (activePage - 1) * itemsCountPerPage,
    activePage * itemsCountPerPage
  );

const sortIconDirection = (direction: Direction) =>
  direction === "asc" ? "sort-down" : "sort-up";

type SortIconProps = {
  column: string;
  orderBy: {
    column: string;
    direction: Direction;
  };
};

const SortIcon: FunctionComponent<SortIconProps> = ({ column, orderBy }) => (
  <FontAwesomeIcon
    icon={
      orderBy.column === column ? sortIconDirection(orderBy.direction) : "sort"
    }
  />
);

type HeaderModifiedProps = {
  onHeaderClick: (column: string, sortFunc: SortFunc) => void;
  // TODO
  sortByCommittedDate: SortFunc;
  orderBy: {
    column: string;
    direction: Direction;
  };
};

const HeaderModified: FunctionComponent<HeaderModifiedProps> = ({
  onHeaderClick,
  sortByCommittedDate,
  orderBy,
}) => (
  <OverlayTrigger
    transition={false}
    overlay={<Tooltip id="last-commit-tooltip">Last commit on master.</Tooltip>}
  >
    <th onClick={() => onHeaderClick("committedDate", sortByCommittedDate)}>
      <FontAwesomeIcon icon="calendar-alt" /> Modified{" "}
      <SortIcon column="committedDate" orderBy={orderBy} />
    </th>
  </OverlayTrigger>
);

type ResultTableProps = {
  forks: Node[];
  activePage: number;
  itemsCountPerPage: number;
  onPageChange: (pageNumber: number) => void;
};

const ResultTable: FunctionComponent<ResultTableProps> = ({
  forks,
  activePage,
  itemsCountPerPage,
  onPageChange,
}) => {
  const sortObjectsFunc =
    (attribute: (attribute: any) => any) =>
    (collection: any[], order: Direction) =>
      lodashOrderBy(collection, [attribute], [order]);
  const sortByNameWithOwner = sortObjectsFunc((x) =>
    x.nameWithOwner.toLowerCase()
  );
  const sortByStargazerCount = sortObjectsFunc((x) => x.stargazerCount);
  const sortByForkCount = sortObjectsFunc((x) => x.forkCount);
  const sortByCommits = sortObjectsFunc((x) => x.object.history.totalCount);
  const sortByCommittedDate = sortObjectsFunc((x) =>
    Date.parse(x.object.committedDate)
  );
  const [orderBy, setOrderBy] = useState<{
    column: string;
    direction: Direction;
    sortFunc: SortFunc;
  }>({
    column: "stargazerCount",
    direction: "desc",
    sortFunc: sortByStargazerCount,
  });
  const onHeaderClick = (column: string, sortFunc: SortFunc) => {
    // change direction only if the same order was selected
    const toggledDirection = orderBy.direction === "desc" ? "asc" : "desc";
    const direction =
      column === orderBy.column ? toggledDirection : orderBy.direction;
    setOrderBy({ column, direction, sortFunc });
  };
  const sortedForks = orderBy.sortFunc(forks.slice(), orderBy.direction);
  return (
    <>
      <Table striped bordered hover>
        <thead>
          <tr>
            <th
              onClick={() =>
                onHeaderClick("nameWithOwner", sortByNameWithOwner)
              }
            >
              <FontAwesomeIcon icon={["fab", "github-alt"]} /> Repo{" "}
              <SortIcon column="nameWithOwner" orderBy={orderBy} />
            </th>
            <th
              onClick={() =>
                onHeaderClick("stargazerCount", sortByStargazerCount)
              }
            >
              <FontAwesomeIcon icon="star" /> Stars{" "}
              <SortIcon column="stargazerCount" orderBy={orderBy} />
            </th>
            <th onClick={() => onHeaderClick("forkCount", sortByForkCount)}>
              <FontAwesomeIcon icon="code-branch" /> Forks{" "}
              <SortIcon column="forkCount" orderBy={orderBy} />
            </th>
            <th onClick={() => onHeaderClick("commits", sortByCommits)}>
              <FontAwesomeIcon icon="dot-circle" /> Commits{" "}
              <SortIcon column="commits" orderBy={orderBy} />
            </th>
            <HeaderModified
              onHeaderClick={onHeaderClick}
              sortByCommittedDate={sortByCommittedDate}
              orderBy={orderBy}
            />
          </tr>
        </thead>
        <tbody>
          {paginatedForks(sortedForks, activePage, itemsCountPerPage).map(
            (fork) => (
              <ForkLine key={fork.nameWithOwner} info={fork} />
            )
          )}
        </tbody>
      </Table>
      <Pagination
        itemClass="page-item"
        linkClass="page-link"
        activePage={activePage}
        itemsCountPerPage={itemsCountPerPage}
        totalItemsCount={forks.length}
        pageRangeDisplayed={5}
        onChange={onPageChange}
      />
    </>
  );
};

export default ResultTable;
