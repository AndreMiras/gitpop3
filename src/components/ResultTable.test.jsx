import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { library } from '@fortawesome/fontawesome-svg-core';
import { fab } from '@fortawesome/free-brands-svg-icons';
import { fas } from '@fortawesome/free-solid-svg-icons';
import ResultTable from './ResultTable';

library.add(fab, fas);

const forks = [
  {
    nameWithOwner: 'django-nonrel/django',
    stargazerCount: 214,
    forkCount: 84,
    object: {
      committedDate: '2020-08-29T14:23:26Z',
      history: {
        totalCount: 13990,
      },
    },
  },
  {
    nameWithOwner: 'django/django',
    stargazerCount: 54330,
    forkCount: 23377,
    object: {
      committedDate: '2020-12-07:14:12Z',
      history: {
        totalCount: 29000,
      },
    },
  },
  {
    nameWithOwner: 'FlipperPA/django-mssql-backend',
    stargazerCount: 18,
    forkCount: 2,
    object: {
      committedDate: '2020-01-13:11:12Z',
      history: {
        totalCount: 28017,
      },
    },
  },
];

test('renders', () => {
  Date.now = jest.fn(() => new Date('2020-12-08T19:18:03.135Z').valueOf());
  const tree = renderer.create(
    <ResultTable forks={forks} activePage={1} itemsCountPerPage={2} onPageChange={() => null} />,
  ).toJSON();
  Date.now.mockRestore();
  expect(tree).toMatchSnapshot();
});

test('sorting', () => {
  render(
    <ResultTable forks={forks} activePage={1} itemsCountPerPage={1} onPageChange={() => null} />,
  );
  // sorted by stargazerCount default
  expect(screen.getByText(forks[1].nameWithOwner)).toBeInTheDocument();
  expect(screen.queryByText(forks[0].nameWithOwner)).not.toBeInTheDocument();
  expect(screen.queryByText(forks[2].nameWithOwner)).not.toBeInTheDocument();
  const repoTableHeader = screen.getByText('Repo');
  // let's sort by repo name
  fireEvent.click(repoTableHeader);
  expect(screen.queryByText('django/django')).not.toBeInTheDocument();
  expect(screen.queryByText('django-nonrel/django')).not.toBeInTheDocument();
  expect(screen.getByText('FlipperPA/django-mssql-backend')).toBeInTheDocument();
  // clicking again should invert it
  fireEvent.click(repoTableHeader);
  expect(screen.queryByText('django/django')).not.toBeInTheDocument();
  expect(screen.getByText('django-nonrel/django')).toBeInTheDocument();
  expect(screen.queryByText('FlipperPA/django-mssql-backend')).not.toBeInTheDocument();
  // same thing for stars
  const starsTableHeader = screen.getByText('Stars');
  // not so sure why it's ascending on first click when it should be descending
  fireEvent.click(starsTableHeader);
  fireEvent.click(starsTableHeader);
  expect(screen.getByText('django/django')).toBeInTheDocument();
  // forks
  const forksTableHeader = screen.getByText('Forks');
  fireEvent.click(forksTableHeader);
  expect(screen.getByText('django/django')).toBeInTheDocument();
  // commits
  const commitsTableHeader = screen.getByText('Commits');
  fireEvent.click(commitsTableHeader);
  expect(screen.getByText('django/django')).toBeInTheDocument();
  // modified date
  const modifiedTableHeader = screen.getByText('Modified');
  fireEvent.click(modifiedTableHeader);
  expect(screen.getByText('django/django')).toBeInTheDocument();
});
