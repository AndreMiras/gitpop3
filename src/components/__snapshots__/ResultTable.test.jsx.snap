// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Array [
  <table
    className="table table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-github-alt fa-w-15 "
            data-icon="github-alt"
            data-prefix="fab"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 480 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M186.1 328.7c0 20.9-10.9 55.1-36.7 55.1s-36.7-34.2-36.7-55.1 10.9-55.1 36.7-55.1 36.7 34.2 36.7 55.1zM480 278.2c0 31.9-3.2 65.7-17.5 95-37.9 76.6-142.1 74.8-216.7 74.8-75.8 0-186.2 2.7-225.6-74.8-14.6-29-20.2-63.1-20.2-95 0-41.9 13.9-81.5 41.5-113.6-5.2-15.8-7.7-32.4-7.7-48.8 0-21.5 4.9-32.3 14.6-51.8 45.3 0 74.3 9 108.8 36 29-6.9 58.8-10 88.7-10 27 0 54.2 2.9 80.4 9.2 34-26.7 63-35.2 107.8-35.2 9.8 19.5 14.6 30.3 14.6 51.8 0 16.4-2.6 32.7-7.7 48.2 27.5 32.4 39 72.3 39 114.2zm-64.3 50.5c0-43.9-26.7-82.6-73.5-82.6-18.9 0-37 3.4-56 6-14.9 2.3-29.8 3.2-45.1 3.2-15.2 0-30.1-.9-45.1-3.2-18.7-2.6-37-6-56-6-46.8 0-73.5 38.7-73.5 82.6 0 87.8 80.4 101.3 150.4 101.3h48.2c70.3 0 150.6-13.4 150.6-101.3zm-82.6-55.1c-25.8 0-36.7 34.2-36.7 55.1s10.9 55.1 36.7 55.1 36.7-34.2 36.7-55.1-10.9-55.1-36.7-55.1z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
           
          Repo
           
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-sort fa-w-10 "
            data-icon="sort"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </th>
        <th
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 "
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
           
          Stars
           
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-sort-up fa-w-10 "
            data-icon="sort-up"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M279 224H41c-21.4 0-32.1-25.9-17-41L143 64c9.4-9.4 24.6-9.4 33.9 0l119 119c15.2 15.1 4.5 41-16.9 41z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </th>
        <th
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-code-branch fa-w-12 "
            data-icon="code-branch"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M384 144c0-44.2-35.8-80-80-80s-80 35.8-80 80c0 36.4 24.3 67.1 57.5 76.8-.6 16.1-4.2 28.5-11 36.9-15.4 19.2-49.3 22.4-85.2 25.7-28.2 2.6-57.4 5.4-81.3 16.9v-144c32.5-10.2 56-40.5 56-76.3 0-44.2-35.8-80-80-80S0 35.8 0 80c0 35.8 23.5 66.1 56 76.3v199.3C23.5 365.9 0 396.2 0 432c0 44.2 35.8 80 80 80s80-35.8 80-80c0-34-21.2-63.1-51.2-74.6 3.1-5.2 7.8-9.8 14.9-13.4 16.2-8.2 40.4-10.4 66.1-12.8 42.2-3.9 90-8.4 118.2-43.4 14-17.4 21.1-39.8 21.6-67.9 31.6-10.8 54.4-40.7 54.4-75.9zM80 64c8.8 0 16 7.2 16 16s-7.2 16-16 16-16-7.2-16-16 7.2-16 16-16zm0 384c-8.8 0-16-7.2-16-16s7.2-16 16-16 16 7.2 16 16-7.2 16-16 16zm224-320c8.8 0 16 7.2 16 16s-7.2 16-16 16-16-7.2-16-16 7.2-16 16-16z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
           
          Forks
           
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-sort fa-w-10 "
            data-icon="sort"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </th>
        <th
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-dot-circle fa-w-16 "
            data-icon="dot-circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm80 248c0 44.112-35.888 80-80 80s-80-35.888-80-80 35.888-80 80-80 80 35.888 80 80z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
           
          Commits
           
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-sort fa-w-10 "
            data-icon="sort"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </th>
        <th
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-calendar-alt fa-w-14 "
            data-icon="calendar-alt"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V192H0v272zm320-196c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM192 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM64 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zM400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v48h448v-48c0-26.5-21.5-48-48-48z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
           
          Modified
           
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-sort fa-w-10 "
            data-icon="sort"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M41 288h238c21.4 0 32.1 25.9 17 41L177 448c-9.4 9.4-24.6 9.4-33.9 0L24 329c-15.1-15.1-4.4-41 17-41zm255-105L177 64c-9.4-9.4-24.6-9.4-33.9 0L24 183c-15.1 15.1-4.4 41 17 41h238c21.4 0 32.1-25.9 17-41z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <a
            href="https://github.com/django-nonrel/django"
          >
            django-nonrel/django
          </a>
        </td>
        <td>
          214
        </td>
        <td>
          84
        </td>
        <td>
          13990
        </td>
        <td>
          3 months ago
        </td>
      </tr>
      <tr>
        <td>
          <a
            href="https://github.com/django/django"
          >
            django/django
          </a>
        </td>
        <td>
          54330
        </td>
        <td>
          23377
        </td>
        <td>
          29000
        </td>
        <td>
          1 days ago
        </td>
      </tr>
    </tbody>
  </table>,
  <ul
    className="pagination"
  >
    <li
      className="page-item disabled"
      onClick={[Function]}
    >
      <a
        aria-label="Go to first page"
        className="page-link"
        href="#"
      >
        «
      </a>
    </li>
    <li
      className="page-item disabled"
      onClick={[Function]}
    >
      <a
        aria-label="Go to previous page"
        className="page-link"
        href="#"
      >
        ⟨
      </a>
    </li>
    <li
      className="page-item active"
      onClick={[Function]}
    >
      <a
        aria-label="Go to page number 1"
        className="page-link undefined"
        href="#"
      >
        1
      </a>
    </li>
    <li
      className="page-item"
      onClick={[Function]}
    >
      <a
        aria-label="Go to page number 2"
        className="page-link"
        href="#"
      >
        2
      </a>
    </li>
    <li
      className="page-item"
      onClick={[Function]}
    >
      <a
        aria-label="Go to next page"
        className="page-link"
        href="#"
      >
        ⟩
      </a>
    </li>
    <li
      className="page-item"
      onClick={[Function]}
    >
      <a
        aria-label="Go to last page"
        className="page-link"
        href="#"
      >
        »
      </a>
    </li>
  </ul>,
]
`;
